systemtest/src/test/java/io/strimzi/systemtest/RollingUpdateST.java

```diff
@@ -912,6 +881,101 @@ void testClusterCaRemovedTriggersRollingUpdate() {
+        String zkMetricsOutput = MetricsUtils.collectMetrics(CLUSTER_NAME + "-zookeeper-0", "");
+        assertThat(kafkaMetricsOutput, containsString("kafka_"));
+        assertThat(zkMetricsOutput, containsString("replicaId"));
+
+        LOGGER.info("Changing metrics to something else");
+        kafkaRule.replace("pattern", "kafka.(\\w+)<type=(.+), name=(.+)><>Count",
+                "kafka.(\\w+)<type=(.+), name=(.+)Percent\\w*><>MeanRate");
+        kafkaRule.replace("name", "kafka_$1_$2_$3_count", "kafka_$1_$2_$3_percent");
+        kafkaRule.replace("type", "COUNTER", "GAUGE");
+
+        zookeeperRule.replace("pattern",
+                "org.apache.ZooKeeperService<name0=ReplicatedServer_id(\\d+), name1=replica.(\\d+)><>(\\w+)",
+                "org.apache.ZooKeeperService<name0=StandaloneServer_port(\\d+)><>(\\w+)");
+        zookeeperRule.replace("name", "zookeeper_$3", "zookeeper_$2");
+        zookeeperRule.replace("labels", zookeeperLabels, null);
+
+        KafkaResource.replaceKafkaResource(CLUSTER_NAME, kafka -> {
+            kafka.getSpec().getKafka().setMetrics(kafkaMetrics);
+            kafka.getSpec().getZookeeper().setMetrics(zookeeperMetrics);
+        });
+
+        LOGGER.info("Check if Kafka and Zookeeper metrics are changed");
+        assertThat(KafkaResource.kafkaClient().inNamespace(NAMESPACE).withName(CLUSTER_NAME).get().getSpec().getKafka().getMetrics(), is(kafkaMetrics));
+        assertThat(KafkaResource.kafkaClient().inNamespace(NAMESPACE).withName(CLUSTER_NAME).get().getSpec().getZookeeper().getMetrics(), is(zookeeperMetrics));
+
+        LOGGER.info("Check if metrics are present in pod of Kafka and Zookeeper");
+
+        kafkaMetricsOutput = MetricsUtils.collectMetrics(KafkaResources.kafkaPodName(CLUSTER_NAME, 0), "");
+        zkMetricsOutput = MetricsUtils.collectMetrics(CLUSTER_NAME + "-zookeeper-0", "");
+
+        assertThat(kafkaMetricsOutput, containsString("kafka_"));
+        assertThat(zkMetricsOutput, containsString("replicaId"));
+
+        Map<String, String> kafkaPods = StatefulSetUtils.ssSnapshot(KafkaResources.kafkaStatefulSetName(CLUSTER_NAME));
+        Map<String, String> zkPods = StatefulSetUtils.ssSnapshot(KafkaResources.zookeeperStatefulSetName(CLUSTER_NAME));
+
+        LOGGER.info("Changing metrics to null");
+
+        KafkaResource.replaceKafkaResource(CLUSTER_NAME, kafka -> {
+            kafka.getSpec().getKafka().setMetrics(null);
+            kafka.getSpec().getZookeeper().setMetrics(null);
+        });
+
+        StatefulSetUtils.waitTillSsHasRolled(KafkaResources.zookeeperStatefulSetName(CLUSTER_NAME), 3, zkPods);
+        StatefulSetUtils.waitTillSsHasRolled(KafkaResources.kafkaStatefulSetName(CLUSTER_NAME), 3, kafkaPods);
+
+        LOGGER.info("Check if metrics is empty in pods");
+
+        kafkaMetricsOutput = MetricsUtils.collectMetrics(KafkaResources.kafkaPodName(CLUSTER_NAME, 0), "");
```

Comment:
I think you can use `collectKafkaPodsMetrics`, same for zookeeper. Or is there any reason why you didn't use it?